core/src/test/java/org/apache/iceberg/MetadataTableScanTestBase.java

```diff
@@ -79,24 +80,19 @@ protected void validateTaskScanResiduals(TableScan scan, boolean ignoreResiduals
     }
   }
 
-  protected void validateIncludesPartitionScan(
-      CloseableIterable<FileScanTask> tasks, int partValue) {
-    validateIncludesPartitionScan(tasks, 0, partValue);
-  }
-
-  protected void validateIncludesPartitionScan(
-      CloseableIterable<FileScanTask> tasks, int position, int partValue) {
+  protected void validatePartition(
+      Iterable<PartitionsTable.Partition> parts, int position, int partitionValue) {
```

Comment:
Id actually prefer this to be CloseableIterable<DataFile>, as Partition seems a bit internal class and shouldn't be used.  I think it would be the same, unless I'm mistaken?